### PR TITLE
improve(index.js): Move --mnemonic to .env file, auto estimate gas price

### PR DESCRIPTION
**Pull request by nicholaspai**
- We shouldn't be encouraging users to enter in their mnemonics on the command line, as this relies on network-level security. Instead, read `MNEMONIC` from the environment or default to the node's unlocked account.
- web3.eth has a nice feature that returns an estimated gas price, we should relax the CLI flag `--gasprice` to be an optional input, and default to this estimated gas price.
- 
Signed-off-by: Nick Pai <npai.nyc@gmail.com>

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -90,10 +98,12 @@ if (argv.gasprice < 1 || argv.gasprice > 1000) throw "--gasprice must be between
     getAddress("PerpetualCreator", networkId)
   );
 
-  // Transaction parameters
+  // Transaction parameters:
+  // - Web3 estimates the gas price using the last few blocks median gas price.
+  const _gasPrice = argv.gasprice ? argv.gasprice * 1000000000 : await web3.eth.getGasPrice();
```
Comment (mrice32) [inline index.js:103]: Does web3.eth.getGasPrice() work on forks, testnets, etc?

File: index.js [hunk 2]
```diff
@@ -102,6 +112,7 @@ if (argv.gasprice < 1 || argv.gasprice > 1000) throw "--gasprice must be between
   const address = await perpetualCreator.methods.createPerpetual(perpetualParams, configSettings).call(transactionOptions);
   console.log("Simulation successful. Expected Address:", address);
 
+  return;
```
Comment (mrice32) [inline index.js]: Early return?

File: index.js [hunk 3]
```diff
@@ -2,32 +2,40 @@ const Web3 = require("web3");
...
+// ---gasprice: gas price to use in GWEI. Defaults to the web3.eth estimated gas price.
+
 const argv = require("minimist")(process.argv.slice(), {
-  string: ["url", "mnemonic", "priceFeedIdentifier", "fundingRateIdentifier", "collateralAddress", "syntheticName", "syntheticSymbol", "minSponsorTokens"]
+  string: ["url", "priceFeedIdentifier", "fundingRateIdentifier", "collateralAddress", "syntheticName", "syntheticSymbol", "minSponsorTokens"],
+  number: ["gasprice"]
 });
 
+// Sanity test arguments:
+if (!argv.url.startsWith("https")) throw "--url must be an HTTPS endpoint";
```
Comment (mrice32) [inline index.js]: Can this not be websocket?

File: index.js [hunk 4]
```diff
@@ -2,32 +2,40 @@ const Web3 = require("web3");
 const HDWalletProvider = require("@truffle/hdwallet-provider");
 const { getAbi, getAddress } = require("@uma/core");
 const { parseFixed } = require("@ethersproject/bignumber");
+require("dotenv").config();
```
Comment (mrice32) [inline index.js:5]: Can you document these changes in the README? It also may impact the docs on the docs site.

File: index.js (hunk 2 above)
Comment (nicholaspai) [reply index.js]: Mistake will remove

File: index.js (hunk 3 above)
Comment (nicholaspai) [reply index.js]: HDWalletProvider seems to expect http

File: index.js (hunk 4 above)
Comment (nicholaspai) [reply index.js:5]: Updated readme to make clear that MNEMONIC should be set as an environment variable

File: index.js (hunk 1 above)
Comment (nicholaspai) [reply index.js:103]: Seems to yep

File: package.json [hunk 5]
```diff
@@ -21,9 +21,10 @@
   },
   "dependencies": {
     "@ethersproject/bignumber": "^5.0.14",
-    "@truffle/hdwallet-provider": "^1.2.1",
-    "@uma/core": "^2.0.1",
+    "@truffle/hdwallet-provider": "^1.2.3",
```
Comment (nicholaspai) [inline package.json:24]: Bumped versions to match `@uma` modules

**Pull request closed without merging**